core/storage/topic/gae_models.py

```diff
@@ -388,6 +383,14 @@ class TopicSummaryModel(base_models.BaseModel):
     # The thumbnail background color of the topic.
     thumbnail_bg_color = datastore_services.StringProperty(indexed=True)
     version = datastore_services.IntegerProperty(required=True)
+    # A mapping whose key is a story id of a published story that belongs to
+    # the topic. Each key maps to a list of exploration ids linked to the
+    # corresponding story's published chapters. This mapping must be kept
+    # up-to-date with changes related to publishing status of stories, creation
```

Comment:
with changes related to ... --> when stories are published/unpublished, story chapters are created or removed, chapters are published/unpublished, and exploration IDs are linked/unlinked to chapters.

Though you could also delete this altogether and say something like "This mapping should be regenerated whenever a story in a topic is updated."

Btw where does this regeneration get triggered? story_services.py doesn't seem to have many changes.

Reply:
The regeneration of the mapping gets triggered in `topic_services.compute_summary_of_topic`. I'll include this in the documentation for this specific field.

Reply:
Done. I went with the latter: deleting all the listed events and summarizing with the above quote you provided.